git-send-email.perl

```diff
@@ -920,7 +935,7 @@ sub validate_address {
 			valid_re => qr/^(?:quit|q|drop|d|edit|e)/i,
 			default => 'q');
 		if (/^d/i) {
-			return undef;
+			return;
```

Comment:
They're 'necessary' to the extent that compliance with Perl Best Practices as enforced by perlcritic is necessary.  You either subscribe to that style of coding or you don't.

A lot of people have gripes with perlcritic: I have some minor issues of my own with it.

So what are we really disagreeing about?  That PBP is a waste of time?  Then let's talk about that.

How are 'return undef' and 'return' different, in your opinion?

As best as I can tell, they are exactly the same semantically, only one conforms to PBP and the other doesn't.

Would parens around:

$x = $y == 10;

be necessary or not?  The answer is, "it depends on what you're trying to do."

If the above expression is inside the condition of an "if", and you're trying to set $x to whether $y is equal to 10 or not, then they aren't necessary.  If you're trying to assign $x the value of $y and then test if that's equal to 10, then they are necessary.  And if you're trying to make the grouping completely unambiguous (as PBP does), then they are necessary (regardless of where they go).

So it's a question of (1) what's the goal and (2) is that a worthwhile goal?